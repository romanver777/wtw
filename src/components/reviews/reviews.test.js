/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from "react";
import TestRenderer from "react-test-renderer";
import Reviews from "./reviews";

describe("Reviews component", () => {
  const props = {
    reviews: [
      {
        reviewId: 3016116,
        reviewType: "POSITIVE",
        reviewData: "2021-04-12T19:01:00",
        userPositiveRating: null,
        userNegativeRating: null,
        reviewAutor: "SlavaLim",
        reviewTitle: "В этом плоде найдется и вкус и приятное послевкусие.",
        reviewDescription:
          "Весьма рекомендую к просмотру фильм с очаровательной Андрой Дэй «Соединённые Штаты против Билли Холидей» (The United States vs. Billie Holiday, 2021) режиссера Ли Дэниелса. \r\nНе относитесь к нему, как к кино-шедевру, но и не сочтите его за грязную пропаганду BLM.",
      },
      {
        reviewId: 3015986,
        reviewType: "POSITIVE",
        reviewData: "2021-04-12T10:26:00",
        userPositiveRating: null,
        userNegativeRating: null,
        reviewAutor: "DIMI25",
        reviewTitle: "Странные плоды",
        reviewDescription:
          "В 1940-х правительство США начинает войну против наркотиков. Ведомые предубеждениями против цветного населения чиновники выбирают своей целью популярную певицу Билли Холидей.\r\nБилли Холидей (англ. Billie Holiday), настоящее имя Элеанора Фейган (англ. Eleanora Fagan; 7 апреля 1915, Балтимор — 17 июля 1959,",
      },
    ],
  };
  const Trenderer = TestRenderer.create(<Reviews {...props} />);

  it("renders correctly", () => {
    expect(Trenderer.toJSON()).toMatchSnapshot();
  });

  it("should find root elem with className 'movie-card__reviews movie-card__row'", () => {
    expect(
      Trenderer.root.findAllByProps({
        className: "movie-card__reviews movie-card__row",
      })
    ).toHaveLength(1);
  });
  it("should find elems with className 'review__text'", () => {
    expect(
      Trenderer.root.findAllByProps({ className: "review__text" })
    ).toHaveLength(2);
  });
  it("should find first review author [props.review[0].reviewAutor]", () => {
    expect(
      Trenderer.root.findAllByProps({ className: "review__author" })[0].children
    ).toEqual([props.reviews[0].reviewAutor]);
  });
  it("should find second review author [props.review[1].reviewAutor]", () => {
    expect(
      Trenderer.root.findAllByProps({ className: "review__author" })[1].children
    ).toEqual([props.reviews[1].reviewAutor]);
  });
});
