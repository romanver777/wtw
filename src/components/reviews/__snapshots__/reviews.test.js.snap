// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews component renders correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Весьма рекомендую к просмотру фильм с очаровательной Андрой Дэй «Соединённые Штаты против Билли Холидей» (The United States vs. Billie Holiday, 2021) режиссера Ли Дэниелса. 
Не относитесь к нему, как к кино-шедевру, но и не сочтите его за грязную пропаганду BLM.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            SlavaLim
          </cite>
          <time
            className="review__date"
            dateTime="2021-04-12T19:01:00"
          >
            12 апреля 2021 г., 19:01
          </time>
        </footer>
      </blockquote>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          В 1940-х правительство США начинает войну против наркотиков. Ведомые предубеждениями против цветного населения чиновники выбирают своей целью популярную певицу Билли Холидей.
Билли Холидей (англ. Billie Holiday), настоящее имя Элеанора Фейган (англ. Eleanora Fagan; 7 апреля 1915, Балтимор — 17 июля 1959,
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            DIMI25
          </cite>
          <time
            className="review__date"
            dateTime="2021-04-12T10:26:00"
          >
            12 апреля 2021 г., 10:26
          </time>
        </footer>
      </blockquote>
    </div>
  </div>
</div>
`;
